### PR TITLE
Use java_common.merge instead of manual _collect util functions in scala_import

### PR DESCRIPTION
**Pull request by long-stripe**
### Description

Reworks some code in `scala_import` to use `java_common.merge` instead of manually creating depsets in utils.

Tests should still pass.

### Motivation

I had remembered seeing this `TODO` before and it seems easy enough to fix. Deleting unneeded code!

7a8

### CODE REVIEW

File: scala/scala_import.bzl [hunk 1]
```diff
@@ -13,9 +13,10 @@ def _scala_import_impl(ctx):
         intellij_metadata,
     ) = (target_data.code_jars, target_data.intellij_metadata)
     current_jars = depset(current_target_compile_jars)
-    exports = _collect(ctx.attr.exports)
-    transitive_runtime_jars = _collect_runtime(ctx.attr.runtime_deps)
-    jars = _collect(ctx.attr.deps)
```
Comment (long-stripe) [inline scala/scala_import.bzl]: I wanted to be sure to call this line out: as it stands in master, this `jars` is unused and is deleted here.

**Comment by johnynek** [comment]
This looks great to me! Thanks for the simplification.

@iirina there were bugs around merge in the past (something with empty lists, or not merging everything about the provider... are those fixed? Do you recall what I am referencing?)

**Comment by iirina** [comment]
This is such a nice change! :) Most bugs in `java_common.merge` are fixed. The only remaining issue is that the _direct_ runtime jars are not merged, but all the transitive closure is. If you have provider A with direct_jars=A.jar and B with direct_jars=B.jar, `java_common.merge(A,B).runtime_output_jars` is an empty list (probably the culprit of https://github.com/bazelbuild/bazel/issues/8916). This PR is only dealing with transitive closure, so all should work as expected.

**Comment by johnynek** [comment]
@iirina thank you for looking!

**Comment by johnynek** [comment]
@ittaiz if you have time will you double check and merge?

**Pull request merged**